Chiazokam/XamBox
# #162932875 Ft api signup

#### What does this PR do?
- sets up middlewares to test user input
- creates a middleware to check if user already exists
- creates the user sign up controller
- creates the root route with welcome message
- creates the signup route and include middlewares
- writes tests to cover all the functions
- sets up CI with travis
- reverts verifyToken to verifytoken in middlewares/index.js
- attempts to set up code climate

#### Description of Task to be completed?
- set up middlewares to test user input
- create a middleware to check if user already exists
- create the user sign up controller
- create the root route with welcome message
- create the signup route and include middlewares
- write tests to cover all the functions
- set up CI with travis
- revert verifyToken to verifytoken in middlewares/index.js
- attempt to set up code climate: revert

#### How should this be manually tested?
- Clone the repo - `https://github.com/Chiazokam/xambox`
- Enter into the project directory - `cd xambox`
- switch to this branch - `checkout ft-api-signup-162932875`
- Start the server `npm run start:dev`
- Open `Postman`
- On `postman` Send a `POST` request to `http://localhost:3000/api/v1/signup`

- You should receive a successful response

#### Any background context you want to provide?
Input data of the format:
`{
                "firstname": "Mayor",
                "lastname": "Franklyn",
                "phone": "+234-7038930045",
                "email": "myfrankly@frank.com",
                "password": "mayorfranklyn"
              }`

#### What are the relevant pivotal tracker stories?
[#162932875](https://www.pivotaltracker.com/story/show/162932875)
